File: src/test/java/com/raychenon/leetcode/coin/CoinChange2Test.java
package com.raychenon.leetcode.coin;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 8/6/2020
 * https://leetcode.com/problems/coin-change-2/
 */
public class CoinChange2Test {

    private static CoinChange2 coinChange2;

    @BeforeClass
    public static void setUp() {
        coinChange2 = new CoinChange2();
    }

    @Test
    public void example1Test() {
        assertEquals(4, coinChange2.change(5, new int[]{1, 2, 5}));
    }

    @Test
    public void example2Test() {
        assertEquals(0, coinChange2.change(3, new int[]{2}));
    }

    @Test
    public void example3Test() {
        assertEquals(1, coinChange2.change(10, new int[]{10}));
    }

}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/coin/CoinChange2Test.kt
package com.raychenon.kotlin.leetcode.coin

import junit.framework.TestCase
import org.junit.Test

/**
 * User: raychenon
 * Date: 8/6/2020
 * https://leetcode.com/problems/coin-change-2/
 */
class CoinChange2Test {


    @Test
    fun example1Test() {
        TestCase.assertEquals(4, CoinChange2.change(5, intArrayOf(1, 2, 5)))
    }

    @Test
    fun example2Test() {
        TestCase.assertEquals(0, CoinChange2.change(3, intArrayOf(2)))
    }

    @Test
    fun example3Test() {
        TestCase.assertEquals(1, CoinChange2.change(10, intArrayOf(10)))
    }
}